dojo/notifications/helper.py

```diff
@@ -129,38 +129,51 @@ def process_notifications(event, notifications=None, *args, **kwargs):
 
 def send_slack_notification(event, user=None, *args, **kwargs):
     from dojo.utils import get_system_setting, get_slack_user_id
+
+    def _post_slack_message(channel):
+        res = requests.request(
+            method='POST',
+            url='https://slack.com/api/chat.postMessage',
+            data={
+                'token': get_system_setting('slack_token'),
+                'channel': channel,
+                'text': create_notification_message(event, user, 'slack', *args, **kwargs)
+            })
+
+        if 'error' in res.text:
+            logger.error("Slack is complaining. See raw text below.")
+            logger.error(res.text)
+
     try:
+        # If the user has slack information on profile and chooses to receive slack notifications
+        # Will receive a DM
         if user is not None:
             if hasattr(user, 'usercontactinfo') and user.usercontactinfo.slack_username is not None:
                 slack_user_id = user.usercontactinfo.slack_user_id
                 if user.usercontactinfo.slack_user_id is None:
-                    # Lookup the slack userid
+                    # Lookup the slack userid the first time, then save it.
                     slack_user_id = get_slack_user_id(
                         user.usercontactinfo.slack_username)
                     slack_user_save = UserContactInfo.objects.get(user_id=user.id)
                     slack_user_save.slack_user_id = slack_user_id
                     slack_user_save.save()
 
-                channel = '@%s' % slack_user_id
+                channel = '@{}'.format(slack_user_id)
+                _post_slack_message(channel)
             else:
-                # user has no slack username, skip
+                logger.error("The user does not have a Slack email address in profile.")
```

Comment:
is this really an error that needs looking into? feels more like a warning to me?

Reply:
maybe even info or debug. I think it will be pretty normal for some users to not enter their slack details because they are already choked to death with notifications ;-)

Reply:
I'll put as info.